internal/storage/memory_test.go

```diff
@@ -73,20 +72,224 @@ func TestMemoryIssuerService(t *testing.T) {
 	issSvc, err := newMemoryIssuerService(config)
 	assert.Nil(t, err)
 
-	for _, testCase := range testCases {
-		testCase := testCase
+	t.Run("Create", func(t *testing.T) {
+		t.Parallel()
+
+		issuer := types.Issuer{
+			TenantID:      tenantID,
+			ID:            "6b0117f8-29e4-49fa-841e-63c52aa27d96",
+			Name:          "Good issuer",
+			URI:           "https://issuer-a27d96.info/",
+			JWKSURI:       "https://issuer.info/jwks.json",
+			ClaimMappings: mappings,
+		}
+
+		testCases := []testingx.TestCase[types.Issuer, *types.Issuer]{
+			{
+				Name:  "Success",
+				Input: issuer,
+				CheckFn: func(t *testing.T, res testingx.TestResult[*types.Issuer]) {
+					assert.Nil(t, res.Err)
+					compareIssuers(t, issuer, *res.Success)
+				},
+			},
+		}
+
+		runFn := func(ctx context.Context, input types.Issuer) testingx.TestResult[*types.Issuer] {
+			iss, err := issSvc.CreateIssuer(context.Background(), input)
+
+			result := testingx.TestResult[*types.Issuer]{
+				Success: iss,
+				Err:     err,
+			}
+
+			return result
+		}
 
-		t.Run(testCase.name, func(t *testing.T) {
-			t.Parallel()
+		testingx.RunTests(context.Background(), t, testCases, runFn)
+	})
 
-			iss, err := issSvc.GetByURI(context.Background(), testCase.input)
+	t.Run("GetByURI", func(t *testing.T) {
```

Comment:
Do you want to update the case names to match the function name e.g.  `GetIssuerByURI`?